drivers/node/aws/aws.go

```diff
@@ -0,0 +1,119 @@
+package aws
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	aws_pkg "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/portworx/torpedo/drivers/node"
+)
+
+const (
+	// DriverName is the name of the aws driver
+	DriverName = "aws"
+)
+
+type aws struct {
+	node.Driver
+	session     *session.Session
+	credentials *credentials.Credentials
+	config      *aws_pkg.Config
+	region      string
+	svc         *ec2.EC2
+	instances   []*ec2.Instance
+}
+
+func (aws *aws) String() string {
+	return DriverName
+}
+
+func (aws *aws) Init(sched string) error {
+	sess := session.Must(session.NewSession())
+	aws.session = sess
+	creds := credentials.NewEnvCredentials()
+	aws.credentials = creds
+	aws.region = os.Getenv("AWS_REGION")
+	if aws.region == "" {
+		return fmt.Errorf("Env AWS_REGION not found")
+	}
+	config := &aws_pkg.Config{Region: aws_pkg.String(aws.region)}
+	config.WithCredentials(creds)
+	aws.config = config
+	svc := ec2.New(sess, config)
+	aws.svc = svc
+	instances := aws.getAllInstances()
+	aws.instances = instances
+	return nil
+}
+
+func (aws *aws) TestConnection(n node.Node, options node.TestConectionOpts) error {
+	return nil
+}
+
+func (aws *aws) RebootNode(n node.Node, options node.RebootNodeOpts) error {
+	var err error
+	instanceID, err := aws.getNodeIDByPrivAddr(n)
+	if err != nil {
+		return &node.ErrFailedToRebootNode{
+			Node:  n,
+			Cause: fmt.Sprintf("failed to get instance ID due to: %v", err),
+		}
+	}
+	//Reboot the instance by its InstanceID
+	rebootInput := &ec2.RebootInstancesInput{
+		InstanceIds: []*string{
+			aws_pkg.String(instanceID),
+		},
+	}
+	_, err = aws.svc.RebootInstances(rebootInput)
+	if err != nil {
+		return &node.ErrFailedToRebootNode{
+			Node:  n,
+			Cause: fmt.Sprintf("failed to reboot instance due to: %v", err),
+		}
+	}
+	return nil
+}
+
+func (aws *aws) ShutdownNode(n node.Node, options node.ShutdownNodeOpts) error {
+	return nil
+}
+
+func (aws *aws) getAllInstances() []*ec2.Instance {
+	params := &ec2.DescribeInstancesInput{}
+	resp, err := aws.svc.DescribeInstances(params)
+	if err != nil {
+		logrus.Errorf("there was an error listing instances in %s. Error: %q", aws.region, err.Error())
+		logrus.Fatal(err.Error())
```

Comment:
No helper routine just kill the test. Helper routines should only return error objects.

So remove logrus and return error. Change the function definition to `getAllInstances() ([]*ec2.Instance, error)`

Reply:
fixed